### PR TITLE
Adding options for ioc main

### PR DESCRIPTION
**Pull request by ZohebShaikh**

### CODE REVIEW

File: src/python/epicscorelibs/ioc.py [hunk 1]
```diff
@@ -73,11 +75,16 @@ def out(msg, *args):
...
+
     iocshRegisterCommon()
 
     out('IOC Starting w/ %s \n', dbs)
-    dbdpath = os.path.join(path.base_path, "dbd")
-    for dbd in [b'base.dbd', b'PVAServerRegister.dbd', b'qsrv.dbd']:
+    if dbLoadDatabase(b'base.dbd',DEFAULT_DBD_PATH.encode(),None):
+        raise RuntimeError('Error loading '+DEFAULT_DBD_PATH)
+    
+    for dbd,dbdpath in extra_dbd_load:
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
    base_dbd_load = (('base.dbd', DEFAULT_DBD_PATH),)
    for dbd, dbdpath in base_dbd_load + tuple(extra_dbd_load):
```

File: src/python/epicscorelibs/ioc.py [hunk 2]
```diff
@@ -73,11 +75,16 @@ def out(msg, *args):
...
         if dbLoadDatabase(dbd, dbdpath.encode(), None):
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
        if dbLoadDatabase(dbd.encode(), dbdpath.encode(), None):
```

File: src/python/epicscorelibs/ioc.py [hunk 3]
```diff
@@ -95,7 +102,7 @@ def out(msg, *args):
     out('IOC Running\n')
 
 
-def main():
+def main(extra_dbd_load=[( b"PVAServerRegister.dbd",DEFAULT_DBD_PATH), (b"qsrv.dbd",DEFAULT_DBD_PATH)],extra_dso_load=[]):
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
pva_dbd_load = (("PVAServerRegister.dbd", DEFAULT_DBD_PATH), ("qsrv.dbd", DEFAULT_DBD_PATH))
pva_dso_load = ("..lib.pvAccessIOC", "..lib.qsrv")
def main(extra_dbd_load=pva_dbd_load, extra_dso_load=pva_dso_load):
```

File: src/python/epicscorelibs/ioc.py [hunk 4]
```diff
@@ -73,11 +75,16 @@ def out(msg, *args):
         sys.stderr.write(msg%args)
         sys.stderr.flush()
 
+    for dso in extra_dso_load:
+        _ = ctypes.CDLL(find_dso(dso), ctypes.RTLD_GLOBAL)
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
        ctypes.CDLL(find_dso(dso), ctypes.RTLD_GLOBAL) 
```

File: src/python/epicscorelibs/ioc.py [hunk 5]
```diff
@@ -21,7 +23,7 @@
 dbRecStd = ctypes.CDLL(path.get_lib("dbRecStd"), mode=ctypes.RTLD_GLOBAL)
 pvAccessIOC = ctypes.CDLL(path.get_lib("pvAccessIOC"), mode=ctypes.RTLD_GLOBAL)
 qsrv = ctypes.CDLL(path.get_lib("qsrv"), mode=ctypes.RTLD_GLOBAL)
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
```

**Review by coretl — CHANGES_REQUESTED** [review]
Can also put the PVA dsos in as optional

**Comment by ZohebShaikh** [comment]
> Can also put the PVA dsos in as optional

I have made the changes for it to work I needed to put a kwargs of package -> epicscorelibs.path
I have tested that it is working 
```
python -m epicscorelibs.ioc -d epicscorelibs/test.db 
IOC Starting w/ [('epicscorelibs/test.db', '')] 
IOC dbd loaded
IOC rRDD
IOC db loaded
Starting iocInit
############################################################################
## EPICS 7.0.7.1-DEV
## Rev. 7.0.7.99.0.2
## Rev. Date 7.0.7.99.0.2
############################################################################
iocRun: All initialization complete
IOC Running
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> 
>>> 
>>> 
now exiting InteractiveConsole...
```
For pvxs :-
```
python -m pvxslibs.ioc -d epicscorelibs/test.db 
IOC Starting w/ [('epicscorelibs/test.db', '')] 
IOC dbd loaded
INFO: PVXS QSRV2 is loaded, permitted, and ENABLED.
IOC rRDD
IOC db loaded
Starting iocInit
############################################################################
## EPICS 7.0.7.1-DEV
## Rev. 7.0.7.99.0.2
## Rev. Date 7.0.7.99.0.2
############################################################################
iocRun: All initialization complete
IOC Running
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> 
now exiting InteractiveConsole...

```

**Review by coretl — APPROVED** [review]

File: src/python/epicscorelibs/ioc.py [hunk 6]
```diff
@@ -19,9 +21,8 @@
     Com = ctypes.CDLL(path.get_lib("Com"), mode=ctypes.RTLD_GLOBAL)
 dbCore = ctypes.CDLL(path.get_lib("dbCore"), mode=ctypes.RTLD_GLOBAL)
 dbRecStd = ctypes.CDLL(path.get_lib("dbRecStd"), mode=ctypes.RTLD_GLOBAL)
-pvAccessIOC = ctypes.CDLL(path.get_lib("pvAccessIOC"), mode=ctypes.RTLD_GLOBAL)
-qsrv = ctypes.CDLL(path.get_lib("qsrv"), mode=ctypes.RTLD_GLOBAL)
 
+DEFAULT_DBD_PATH =os.path.join(path.base_path, "dbd")
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
DEFAULT_DBD_PATH = os.path.join(path.base_path, "dbd")
```

File: src/python/epicscorelibs/ioc.py [hunk 7]
```diff
@@ -73,12 +74,16 @@ def out(msg, *args):
...
     iocshRegisterCommon()
 
     out('IOC Starting w/ %s \n', dbs)
-    dbdpath = os.path.join(path.base_path, "dbd")
-    for dbd in [b'base.dbd', b'PVAServerRegister.dbd', b'qsrv.dbd']:
-        if dbLoadDatabase(dbd, dbdpath.encode(), None):
+    
+    base_dbd_load = (('base.dbd',DEFAULT_DBD_PATH),)
+    for dbd,dbdpath in base_dbd_load + tuple(extra_dbd_load):
+        if dbLoadDatabase(dbd.encode(), dbdpath.encode(), None):
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
    base_dbd_load = (('base.dbd', DEFAULT_DBD_PATH),)
    for dbd, dbdpath in base_dbd_load + tuple(extra_dbd_load):
        if dbLoadDatabase(dbd.encode(), dbdpath.encode(), None):
```

File: src/python/epicscorelibs/ioc.py [hunk 8]
```diff
@@ -94,8 +99,9 @@ def out(msg, *args):
         raise RuntimeError('Error starting IOC')
     out('IOC Running\n')
 
-
-def main():
+pva_dbd_load = (("PVAServerRegister.dbd",DEFAULT_DBD_PATH), ("qsrv.dbd",DEFAULT_DBD_PATH))
+pva_dso_load = ("epicscorelibs.lib.pvAccessIOC", "epicscorelibs.lib.qsrv",)
+def main(extra_dbd_load=pva_dbd_load,extra_dso_load=pva_dso_load):
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
pva_dbd_load = (("PVAServerRegister.dbd", DEFAULT_DBD_PATH), ("qsrv.dbd", DEFAULT_DBD_PATH))
pva_dso_load = ("epicscorelibs.lib.pvAccessIOC", "epicscorelibs.lib.qsrv",)
def main(extra_dbd_load=pva_dbd_load, extra_dso_load=pva_dso_load):
```

File: src/python/epicscorelibs/ioc.py [hunk 9]
```diff
@@ -106,7 +112,7 @@ def __call__(self, parser, ns, values, opt):
     parser.add_argument('-d', '--database', default=[], action=DbAction,
         help="Path to database file to load")
     args = parser.parse_args()
-    start_ioc(dbs=args.database)
+    start_ioc(dbs=args.database,extra_dbd_load=extra_dbd_load,extra_dso_load=extra_dso_load)
```
Comment (coretl) [inline src/python/epicscorelibs/ioc.py]:
```suggestion
    start_ioc(dbs=args.database, extra_dbd_load=extra_dbd_load, extra_dso_load=extra_dso_load)
```

**Review by coretl — COMMENTED** [review]
Whitespace changes then done

**Review by coretl — APPROVED** [review]

**Pull request merged**